content/concepts/consensus-network/negative-unl.md

```diff
@@ -0,0 +1,182 @@
+
+Therefore, if more than 20% of trusted validators go offline or become unable to communicate with the rest of the network, the network stops making forward progress because it cannot reach a quorum. This is a design choice to ensure that no transactions' outcomes can be changed after they are declared final. During such a situation, the remaining servers would still be online and could provide past and tentative transaction data, but could not confirm the final, immutable outcome of new transactions.
+
+However, this means that the network could stop making forward progress if a few widely-trusted validators went offline. As of 2020-10-06, there are 34 validators in Ripple's recommended UNL, so the network would stop making forward progress if 7 or more of them were offline. Furthermore, if one or two validators are out for an extended period of time, the network has less room for disagreement between the remaining validators, which can make it take longer to achieve a consensus.
+
+## Enabling the Negative UNL for Testing
+
+Negative UNL functionality is currently available for testing on [Devnet](parallel-networks.html). You can test the Negative UNL feature by adding or modifying a `[features]` stanza in your `rippled.cfg` file, as described in [Connect Your rippled to a Parallel Network](connect-your-rippled-to-the-xrp-test-net.html).
+
+
+## Summary
+
+It is not reasonable to expect a diverse set of validators to maintain 100% uptime: many things can cause a validator to become temporarily unavailable, such as: hardware maintenance, software upgrades, internet connectivity problems, targeted attacks, human error, hardware failures, and outside circumstances like natural disasters.
+
+The "Negative UNL" is **a list of trusted validators which are believed to be offline or malfunctioning**, as declared by a consensus of the remaining validators. Validators in the Negative UNL are ignored for determining if a new ledger version has attained a consensus.
+
+When a validator that is on the Negative UNL comes back online and sends consistent validation votes, the remaining validators remove it from the Negative UNL after a short time.
+
+In cases where validators go offline one or two at a time, the remaining validators can use the Negative UNL to gradually adjust their effective UNLs, so that the network only ever needs 80% of the _online_ validators to achieve a quorum. To prevent the network from fragmenting, the quorum has a hard minimum of 60% of _total_ validators.
+
+The Negative UNL does not help against situations where greater than 20% of trusted validators suddenly go offline all at once. In that case, the remaining validators would not be able to reach the consensus necessary to adjust the Negative UNL.
```

Comment:
I think as long as the network can keep closing ledgers, the effective UNL will be adjusted eventually. 

Say 3 of the 10 validators of a network goes offline at the same time. The network will stop fully validating ledgers. But the network will continue close ledgers with new client txns and pseudo-txns. Hence a UNLModify tx could be added to closed flag ledgers. Once in a closed flag ledger N, the scheduled Negative UNL modification will take effect at the next flag ledger N+256, no matter if N is fully validated or not. So given enough time, the effective quorum will be adjusted. At that time, the network will resume fully validating ledgers.

Reply:
Changed this and some other bits to clarify how the Negative UNL doesn't affect the transaction-set-choosing step of consensus.